### DISCUSSION TITLE
i would like an `on_select` instead of an `on_change` event for the `ui.select` element.

### DISCUSSION THREAD

**Discussion by metaperl** [zauberzeug/nicegui#1611] [Q&A]
### Question

In [my application](https://github.com/metaperl/nvc4me), I have a large number of select elements that all have `on_change=lambda e: add_feeling(e.value)` - https://github.com/metaperl/nvc4me/blob/main/main.py#L36

however, the `on_change` element does not fire when I click the pulldown and select the currently selected element. 

But my desire is that any selected element, including the currently selected element gets added to a set of values when the user chooses the value.

In other words, it does not matter if re-selecting an element does not imply a change. What matters is if the user clicked on the pulldown and then clicked on the element... at which point an (imaginary) `on_select` event would run and add the selected value to a Python set.

This video demonstrates why I want an `on_select` element. - https://www.youtube.com/watch?v=JJ1cNEC7o9E

**Comment by natankeddem** [comment]
You are trying to have all 20 feeling filled in from the start? Like:
`I am feeling compassionate feeling0, feeling1, feeling2, feeling2, ....., feeling19`
Is that correct? 

If that isn't correct. You might also want to try it without choosing any value in the instantiation:
`ui.select(_, on_change=lambda e: add_feeling(e.value))`
That will force the user to pull down the menu and select something.

Going off on a tangent, it seems like this is something suited to a search-as-you-type/multiselect, perhaps?
https://nicegui.io/documentation/select#multi_selection
https://nicegui.io/documentation/select#search-as-you-type

**Reply by metaperl** [reply]
> You are trying to have all 20 feeling filled in from the start?

No. This video demonstrates what I'm looking for - https://www.youtube.com/watch?v=JJ1cNEC7o9E

**Reply by metaperl** [reply]
> If that isn't correct. You might also want to try it without choosing any value in the instantiation:

You've got a point. I could make the first item a category label and have no value selected.

**Comment by falkoschindler** [comment]
Maybe you can use the "popup-hide" event:

```py
select = ui.select(['A', 'B', 'C'], value='A')
select.on('popupHide', lambda: ui.notify(f'Loading {select.value}...'))
```

**Reply by falkoschindler** [reply]
See https://quasar.dev/vue-components/select#qselect-api for a reference of events provided by QSelect.
